src/state.cpp

```diff
@@ -148,6 +148,44 @@ void pop_all()
     TRACE_FUNC_END;
 }
 
+bool check_states(const states::StateId state_given, const states::StateId state_to_check)
```

Comment:
Yes, I will chop down the helper function, it isn't needed.

As for the recursion, I found it more comfortable to write.